### PR TITLE
#718 resolved. Added color to selected item in drawer.

### PR DESCRIPTION
**Pull request by getsanjeev**
#718 fixed. Added color to selected item in drawer.

### CODE REVIEW

File: collect_app/src/main/java/org/odk/collect/android/activities/AppListActivity.java [hunk 1]
```diff
@@ -67,6 +69,8 @@
     protected String[] mSortingOptions;
 
     private boolean mIsSearchBoxShown;
+    private boolean mFirstTime = true;
+    private int moldSelected;
```
Comment (dcbriccetti) [inline collect_app/src/main/java/org/odk/collect/android/activities/AppListActivity.java]: “mold” (a hollow container used to give shape to molten or hot liquid material ...) or `mOld`?

File: collect_app/src/main/java/org/odk/collect/android/activities/AppListActivity.java [hunk 2]
```diff
@@ -189,6 +193,11 @@ private void setupDrawerItems() {
             public View getView(int position, View convertView, ViewGroup parent) {
                 TextView textView = (TextView) super.getView(position, convertView, parent);
                 textView.setPadding(50, 0, 0, 0);
+                if ((position == 0)&&(mFirstTime)) {
```
Comment (dcbriccetti) [inline collect_app/src/main/java/org/odk/collect/android/activities/AppListActivity.java]: The parens around mFirstTime add no value.

File: collect_app/src/main/java/org/odk/collect/android/fragments/AppListFragment.java [hunk 3]
```diff
@@ -60,7 +62,8 @@
     protected SimpleCursorAdapter mListAdapter;
     protected LinkedHashSet<Long> mSelectedInstances = new LinkedHashSet<>();
     protected EditText mInputSearch;
-
+    private int moldSelected;
```
Comment (dcbriccetti) [inline collect_app/src/main/java/org/odk/collect/android/fragments/AppListFragment.java]: I didn’t study this carefully, but is it necessary to have duplicate code here and in AppListActivity?

File: collect_app/src/main/java/org/odk/collect/android/activities/AppListActivity.java (hunk 1 above)
Comment (getsanjeev) [reply collect_app/src/main/java/org/odk/collect/android/activities/AppListActivity.java]: Sorry. I will change it. Yes it should be mOldSelected.

File: collect_app/src/main/java/org/odk/collect/android/activities/AppListActivity.java (hunk 2 above)
Comment (getsanjeev) [reply collect_app/src/main/java/org/odk/collect/android/activities/AppListActivity.java]: certainly they add no functional value. I just feel it makes the code easier to read.

Comment (dcbriccetti) [reply collect_app/src/main/java/org/odk/collect/android/activities/AppListActivity.java]: By “add no value”, I was including readability. I see superfluous symbols as noise that slows down reading. But reasonable people may disagree.

Comment (dcbriccetti) [reply collect_app/src/main/java/org/odk/collect/android/activities/AppListActivity.java]: I would write it
`if (position == 0 && mFirstTime)`
But not everyone remembers that `==` binds tighter than `&&`, so sometimes parentheses aid readability.

File: collect_app/src/main/java/org/odk/collect/android/fragments/AppListFragment.java (hunk 3 above)
Comment (getsanjeev) [reply collect_app/src/main/java/org/odk/collect/android/fragments/AppListFragment.java]: We have called both files, one in all activities and other in 'delete saved form fragment'. Thats why.

File: collect_app/src/main/java/org/odk/collect/android/activities/AppListActivity.java (hunk 1 above)
Comment (getsanjeev) [reply collect_app/src/main/java/org/odk/collect/android/activities/AppListActivity.java]: @dcbriccetti Do I need to correct it and resend PR.

Comment (dcbriccetti) [reply collect_app/src/main/java/org/odk/collect/android/activities/AppListActivity.java]: Just commit and push. The pull request will reflect the most recent versions.

**Comment by grzesiek2010** [comment]
@getsanjeev have you seen [this comment](https://github.com/opendatakit/collect/issues/718#issuecomment-288397050)? It's in progress and it's not yours. The second thing is that this pr solve only a small part of functionality that should be implemented. Sorry but you shouldn't start work on an issue that is already assigne. There are 195 other you can take.
@lognaturel

**Comment by getsanjeev** [comment]
@grzesiek2010  As you can see I asked a clarification as I had already started working on it just when issue was reported. Actually I didnt see you replying in positive when you were mentioned and its been 6 days since then.
Sorry I should have minded that.

**Comment by lognaturel** [comment]
@getsanjeev @grzesiek2010 Yes, let's make sure we are better at saying what's in progress and not taking on projects that are already claimed!

In this case, I think it's ok to split it into two parts -- indicating what the current sort is (this) and remembering the sort for each screen (what @grzesiek2010 will do).

**Comment by getsanjeev** [comment]
@lognaturel Please review.

**Comment by lognaturel** [comment]
@getsanjeev We just merged in the other part of this issue! That means sort order is now saved on a per-screen basis. Could you please merge in those changes?

**Comment by getsanjeev** [comment]
@lognaturel Yes I will do it in a while.

**Comment by lognaturel** [comment]
Replaced by #845

**Pull request closed without merging**